### PR TITLE
Split docs build into travis stage

### LINKED ISSUE #2086: Split doc site build into separate travis job
This will help us avoid build timeouts by moving unrelated tasks into separate jobs (which each get 50 min max). Also when the site fails, it's easier to tell that it's not "a real error". 😂

### PR DESCRIPTION
**Pull request by carolynvs**
This helps us shave time off of the main build (which is limited to 50min) and
makes it easier to identify when the build failed, vs. weird jekyll stuff.

Closes #2086

### CODE REVIEW

**Comment by carolynvs** [comment]
Woot, shaved 2 minutes off of the main build job! Every minute counts! 👍

**Comment by carolynvs** [comment]
Doh! Thanks for the reality check. 😊

**Comment by MHBauer** [comment]
do we need verify-docs to depend on docs?

**Comment by carolynvs** [comment]
Doug's link checker doesn't handle the markdown files right at the moment (as I had origianlly hoped when I made that change). I'll remove that.

**Comment by MHBauer** [comment]
I think I don't know what `make docs` does.

**Comment by carolynvs** [comment]
make docs runs the static site generate on the svc-cat.io markdown files (in /docs and /docsite)

**Comment by carolynvs** [comment]
Sorry for the churn on this. I thought I understand what was going on with the travis config and makefile but turns out there was more to be learned. I'll keep testing and tweaking and report back when I have something solid.

File: Makefile [hunk 1]
```diff
@@ -235,7 +235,7 @@ verify: .init verify-generated verify-client-gen verify-docs verify-vendor
 	@echo Running tag verification:
 	@$(DOCKER_CMD) build/verify-tags.sh
 
-verify-docs: .init docs
+verify-docs: .init
```
Comment (MHBauer) [inline Makefile]: I think this makes sense.

File: Makefile [hunk 2]
```diff
@@ -202,8 +202,8 @@ $(BINDIR):
 
 # Util targets
 ##############
-.PHONY: verify verify-generated verify-client-gen verify-docs
-verify: .init verify-generated verify-client-gen verify-docs verify-vendor
+.PHONY: verify verify-generated verify-client-gen
+verify: .init verify-generated verify-client-gen verify-vendor
```
Comment (MHBauer) [inline Makefile]: why do we want to take verify-docs out of verify-everything?

File: Makefile [hunk 3]
```diff
@@ -202,8 +202,8 @@ $(BINDIR):
 
 # Util targets
 ##############
-.PHONY: verify verify-generated verify-client-gen verify-docs
-verify: .init verify-generated verify-client-gen verify-docs verify-vendor
+.PHONY: verify verify-generated verify-client-gen
```
Comment (MHBauer) [inline Makefile]: Definitely want to leave it in phony, we aren't producing a file `verify-docs`.

**Comment by MHBauer** [comment]
This is how I think it should behave:
DOCS_ONLY
    stage test -> make verify-docs (but no make docs)
    stage docs -> make docs
else
    stage test -> make verify-all (but no make docs)
    stage docs -> make docs

stages go in order. I think maybe `verify-docs` and `docs` make targets don't interact so it's okay they're out-of-order when we look at it in travis.

File: Makefile (hunk 2 above)
Comment (carolynvs) [reply Makefile]: That's was a mistake, I thought I reverted that change, let me try again.

**Comment by carolynvs** [comment]
I am starting to wonder if this doesn't make sense, especially after we move to Hugo (which is much faster than Jekyll). If this doesn't take off enough time, then I'll redo this to just add more relevant files to the docs regex, and keep the fix to the prereqs for verify-docs to not call docs.

**Comment by carolynvs** [comment]
Spoke with @MHBauer on slack and we'd like to make better use of stages to improve build troubleshooting, so I'm going to keep this change as-is.

**Comment by carolynvs** [comment]
rebased. PTAL

**Comment by carolynvs** [comment]
I've rebased and changed from a "docs stage" to a "docs job" so that it runs in parallel. @MHBauer PTAL

**Comment by MHBauer** [comment]
seems right, yes.
